src/pages/CoinListPage/CoinListPage.tsx

```diff
@@ -1,114 +1,100 @@
+import { observer } from "mobx-react-lite";
+import { useNavigate } from "react-router-dom";
 
 import styles from "./CoinListPage.module.scss";
 import CoinList from "./components/CoinList";
 
 const CoinListPage = () => {
-  const { search, handleInput, handlerSearchButton } = useSearchInput();
+  const store = useLocalStore(() => new coinListStore());
+  useAsync(store.fetch, [store.dropdownStore.dropdownValues]);
 
-  const [coins, setCoins] = useState<Coin[]>([]);
-
-  const [coinInfoIsLoading, setCoinInfoIsLoading] = useState(false);
-
-  const [currencyInfoIsLoading, setCurrencyInfoIsLoading] = useState(false);
+  const handlerDropdown = useCallback(
+    (args: any) => {
+      store.dropdownStore.dropdownValues = args;
+    },
+    [store.dropdownStore]
+  );
 
-  const { options, setOptions, dropdownValue, handlerDropdown } = useDropdown({
-    option: [],
-    value: {
-      key: "usd",
-      value: "Market- USD",
+  const handleInput = useCallback(
+    (args: any) => {
+      store.searchStore.search = args.target.value;
     },
-  });
+    [store.searchStore]
+  );
+  const handleButton = useCallback(() => {
+    store.searchStore.search = "";
+  }, [store.searchStore]);
 
-  /* Запрос данных для выпадающего списка*/
-  useEffect(() => {
-    setCurrencyInfoIsLoading(true);
-    axios
-      .get("https://api.coingecko.com/api/v3/simple/supported_vs_currencies")
-      .then((res) => {
-        const data = options;
-        res.data.sort().forEach((element: string, index: number) => {
-          data[index] = {
-            key: element,
-            value: `Market- ${element.toUpperCase()}`,
-          };
-        });
-        setOptions(data);
-        setCurrencyInfoIsLoading(false);
-      })
-      .catch((error) => alert(error));
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  const navigate = useNavigate();
 
   useEffect(() => {
-    setCoinInfoIsLoading(true);
-
-    axios
-      .get(
-        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=${dropdownValue.key}&order=market_cap_desc&per_page=100&page=1&sparkline=true`
-      )
-      .then((res) => {
-        setCoins(res.data);
-        setCoinInfoIsLoading(false);
-      })
-      .catch((error) => alert(error));
-  }, [dropdownValue]);
-
-  const searchedCoins: Coin[] = coins.filter((coin: Coin) => {
-    return coin.name.toLowerCase().includes(search.toLowerCase());
-  });
+    const params = new URLSearchParams();
+    if (store.searchStore._search) {
+      params.append("search", store.searchStore._search);
+    } else {
+      params.delete("search");
+    }
+    if (store.dropdownStore.dropdownValues.key !== "usd") {
+      params.append("currency", store.dropdownStore.dropdownValues.key);
+    } else {
+      params.delete("currency");
+    }
+    if (store.paginationStore.page !== 1) {
+      params.append("page", String(store.paginationStore.page));
+    } else {
+      params.delete("page");
+    }
```

Comment:
`else`-ы тут не нужны. ты создаешь новый пустой `URLSearchParams `, если `!store.searchStore._search`, то ты удаляешь `search `, хотя даже его не аппендил...
```suggestion
    if (store.searchStore._search) {
      params.append("search", store.searchStore._search);
    }
    
    if (store.dropdownStore.dropdownValues.key !== "usd") {
      params.append("currency", store.dropdownStore.dropdownValues.key);
    }
    
    if (store.paginationStore.page !== 1) {
      params.append("page", String(store.paginationStore.page));
    }
```